server/src/test/java/io/crate/integrationtests/LuceneQueryBuilderIntegrationTest.java

```diff
@@ -343,6 +343,66 @@ public void testWhereNotEqualAnyWithLargeArray() throws Exception {
         assertThat(response.rows()[0][0]).isEqualTo(1L);
     }
 
+    @Test
+    public void testWhereNotEqualAnyWithLargeArrayWithIndexOffForIntegerType() throws Exception {
+        // Test overriding of default value 8192 for indices.query.bool.max_clause_count
+        execute("create table t1 (id integer index off) clustered into 2 shards with (number_of_replicas = 0)");
+        execute("create table t2 (id integer index off) clustered into 2 shards with (number_of_replicas = 0)");
+        ensureYellow();
+
+        int bulkSize = NUMBER_OF_BOOLEAN_CLAUSES;
+        Object[][] bulkArgs = new Object[bulkSize][];
+        for (int i = 0; i < bulkSize; i++) {
+            bulkArgs[i] = new Object[]{i};
+        }
+        execute("insert into t1 (id) values (?)", bulkArgs);
+        execute("insert into t2 (id) values (1)");
+        execute("refresh table t1, t2");
+
+        execute("select count(*) from t2 where id != any(select id from t1)");
+        assertThat(response.rows()[0][0]).isEqualTo(1L);
```

Comment:
This test fails with:

```

org.elasticsearch.common.io.stream.NotSerializableExceptionWrapper: too_many_nested_clauses: Query contains too many nested clauses; maxClauseCount is set to 10000

	at __randomizedtesting.SeedInfo.seed([645CB1B2CE098AFF:2F09918313BC6841]:0)
	at org.apache.lucene.search.IndexSearcher$3.visitLeaf(IndexSearcher.java:836)
	at org.apache.lucene.search.FieldExistsQuery.visit(FieldExistsQuery.java:93)
	at org.apache.lucene.search.BooleanQuery.visit(BooleanQuery.java:583)
	at org.apache.lucene.search.IndexSearcher.rewrite(IndexSearcher.java:804)
	at io.crate.execution.engine.collect.collectors.LuceneBatchIterator.createWeight(LuceneBatchIterator.java:188)
	at io.crate.execution.engine.collect.collectors.LuceneBatchIterator.moveNext(LuceneBatchIterator.java:108)
	at io.crate.data.BatchIterator.move(BatchIterator.java:156)
	at io.crate.data.BatchIterators.collect(BatchIterators.java:59)
	at io.crate.data.BatchIterators.collect(BatchIterators.java:42)
	at io.crate.data.CollectingBatchIterator.lambda$newInstance$2(CollectingBatchIterator.java:87)
	at io.crate.data.CollectingBatchIterator.loadNextBatch(CollectingBatchIterator.java:146)
	at io.crate.execution.engine.distribution.DistributingConsumer.consumeIt(DistributingConsumer.java:129)
	at io.crate.execution.engine.distribution.DistributingConsumer.accept(DistributingConsumer.java:105)
	at io.crate.execution.engine.collect.CollectTask.lambda$new$0(CollectTask.java:104)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.lang.Thread.run(Thread.java:1589)

```

should be fixed by 3fa060c66d271d01116effff281e02be87b4df53.